transaction/tests/CycleDetector_unittest.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "transaction/CycleDetector.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "transaction/DirectedGraph.hpp"
+#include "transaction/StronglyConnectedComponents.hpp"
+#include "transaction/Transaction.hpp"
+
+#include "gtest/gtest.h"
+
+namespace quickstep {
+namespace transaction {
+
+class CycleDetectorTest : public testing::Test {
+ protected:
+  const std::uint64_t kNumberOfTransactions = 12;
+  virtual void SetUp() {
+    for (std::uint64_t i = 0; i < kNumberOfTransactions; ++i) {
+      transactions_.push_back(new transaction_id(i));
+    }
+
+    wait_for_graph_ = std::make_unique<DirectedGraph>();
+
+    for (std::uint64_t i = 0; i < kNumberOfTransactions; ++i) {
+      node_ids_.push_back(wait_for_graph_->addNodeUnchecked(transactions_[i]));
+    }
+  }
+
+  std::vector<std::pair<DirectedGraph::node_id, DirectedGraph::node_id>> edges_;
+  std::unique_ptr<DirectedGraph> wait_for_graph_;
+  std::vector<transaction_id*> transactions_;
```

Comment:
Since `transactions_` is not used in other place except for `SetUp`, we could declare it as a local variable.